### ISSUE TITLE
Decide what and how to announce when a file is chosen

### ISSUE THREAD

**Issue by selfthinker** [alphagov/govuk-frontend#5622]
## What

With the native file upload input, not all screen readers always announce the chosen file: JAWS and NVDA both with Chrome and Firefox, VO on macOS with Safari (Chrome seems to be fine).

Our solution currently doesn't announce the file name. But it probably should.

What it could say:

* just the file name
* the file name plus something like "selected"
* the file name plus the page title (because the user had left the browser window just before, but it might say that anyway)
* the file name plus the page title plus the input label (but it might say that anyway)
* something else?

When using the native file input with a screen reader it usually reads quite a lot of things (I think because it got focus back to the browser window) which is quite verbose. It's not a good idea to emulate a specific screen reader directly because they all say wildly different things. It makes sense to say the gist of what they all say.

## Why

To let screen reader users know they have successfully selected a file and which file it was.
Also to meet WCAG 2.2 SC 4.1.3 Status Messages.

## Who needs to work on this

Accessibility specialist, maybe a content designer

## Who needs to review this

We'll review the decision as a team in a decision log meeting

## Done when

- [ ] Options with their pros and cons documented
- [x] Researched what other implementations do
- [x] Decision made and logged

**Comment by selfthinker** [comment]
Last week I've checked how some other design systems's file input works in assistive technologies (AT) and documented it in the [testing spreadsheet](https://docs.google.com/spreadsheets/d/12qUQjVR19092-1t7S-8J3VDArczgs4XW9huxftRoSgc/edit?usp=sharing) (in the 'other DSs' tab).

Based on that and how the native file input works across AT, I think it would be best if we could announce all three relevant elements:
* the file name
* the label
* the button text

But because different screen readers don't always read out everything and they sometimes use a different order, we shouldn't force them to read out everything and in a specific order. We should just make sure that these parts are all connected so they *can* read them all.

The most important thing is that at least the file name is read out as that is the main feedback that the selection was successful.

**Comment by selfthinker** [comment]
We'll discuss this in our decision log meeting today.
